// Copyright 2025 The Ray Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//  http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.


syntax = "proto3";

import "google/protobuf/timestamp.proto";
import "src/ray/protobuf/runtime_env_common.proto";

package ray.rpc.events;

// Message containing the definition information of a driver job.
// The message is expected to be emitted once per job creation.
//
// For runtime information associated with this event, see DriverJobExecutionEvent.
message DriverJobDefinitionEvent {
  message Config {
    RuntimeEnvInfo runtime_env_info = 1;
    map<string, string> metadata = 2;
  }

  bytes job_id = 1;
  int64 driver_pid = 3;
  bytes driver_node_id = 4;
  string entrypoint = 5;
  Config config = 6;
}
